fdbclient/ServerKnobs.cpp

```diff
@@ -443,7 +443,7 @@ void ServerKnobs::initialize(Randomize randomize, ClientKnobs* clientKnobs, IsSi
 	init( BACKUP_TIMEOUT,                                        0.4 );
 	init( BACKUP_NOOP_POP_DELAY,                                 5.0 );
 	init( BACKUP_FILE_BLOCK_BYTES,                       1024 * 1024 );
-	init( BACKUP_LOCK_BYTES,                                     3e9 ); if(randomize && BUGGIFY) BACKUP_LOCK_BYTES = deterministicRandom()->randomInt(1024, 4096) * 1024;
+	init( BACKUP_LOCK_BYTES,                                     3e9 ); if(randomize && BUGGIFY) BACKUP_LOCK_BYTES = deterministicRandom()->randomInt(1024, 4096) * 15 * 1024;
```

Comment:
I chose this number to be at least 15 MB based on the max size of a transaction. I don't know if that's a reasonable bound, so if anyone has a better suggestion I can change it. It should result in many fewer failures, and with the below ASSERT any additional failures should be easier to resolve.

Reply:
I intentionally pick a small buffer size for buggify in order to easily surface memory pressure in simulation. However, this has the unintended hanging as you discovered. I think 15 MB will make that scenario highly unlikely  ---- a version's data above 15MB probably never happens in simulation.